tests/testsuite.src/used_binaries.at

```diff
@@ -1002,3 +1002,125 @@ AT_CHECK([$COBC -fdiagnostics-plain-output -fdiagnostics-show-caret -Wno-others
 
 AT_CLEANUP
 
+
+AT_SETUP([output dependencies])
+# AT_KEYWORDS([])
+
+AT_DATA([prog.cob], [
+        IDENTIFICATION DIVISION.
+        PROGRAM-ID. prog.
+        DATA DIVISION.
+        WORKING-STORAGE SECTION.
+	COPY COPY1.
+        PROCEDURE DIVISION.
+        MAIN-PROC SECTION.
+        00.
+        COPY COPY2.
+        END-PROC  SECTION.
+        COPY COPY3 in "sub".
+        EX.
+            STOP RUN.
+])
+
+AT_CHECK([mkdir -p sub])
+AT_DATA([COPY1.CPY], [])
+AT_DATA([COPY2.CPY], [])
+AT_DATA([sub/COPY3.CPY], [])
+
+AT_CHECK([$COMPILE_ONLY prog.cob])
+
+AT_CHECK([$COMPILE_ONLY -M prog.cob prog.cob], [0],
+[prog.o: \
+ prog.cob \
+ COPY1.CPY \
+ COPY2.CPY \
+ sub/COPY3.CPY
+
+prog.o: \
+ prog.cob \
+ COPY1.CPY \
+ COPY2.CPY \
+ sub/COPY3.CPY
+
+])
+
+AT_CHECK([$COMPILE_ONLY -M -MF prog.dep prog.cob])
+
+AT_CHECK([cat prog.dep], [0],
+[prog.o: \
+ prog.cob \
+ COPY1.CPY \
+ COPY2.CPY \
+ sub/COPY3.CPY
+
+])
+
+AT_CHECK([$COMPILE_ONLY -M -fcopybook-deps prog.cob], [0],
+[prog.o: COPY1 COPY2 sub/COPY3
+
+])
+
+AT_CHECK([$COMPILE_ONLY -M -MT prog.so prog.cob], [0],
+[prog.so: \
+ prog.cob \
+ COPY1.CPY \
+ COPY2.CPY \
+ sub/COPY3.CPY
+
+])
+
+AT_CHECK([$COMPILE_ONLY -M -MQ '$(target)#toto' prog.cob], [0],
+[$$(target)\#toto: \
+ prog.cob \
+ COPY1.CPY \
+ COPY2.CPY \
+ sub/COPY3.CPY
+
+])
+
+AT_CHECK([rm -f prog.d])
+
+AT_CHECK([$COMPILE -MD  prog.cob])
+
+AT_CHECK([cat prog.d], [0],
+[prog.o: \
+ prog.cob \
+ COPY1.CPY \
+ COPY2.CPY \
+ sub/COPY3.CPY
+
+])
+
```

Comment:
`-MD` is for "creating it along" - so please check that the target executable is created here